modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -306,6 +309,11 @@ public void processOpts() {
             this.setRemoveEnumValuePrefix(Boolean.valueOf(additionalProperties
                     .get(CodegenConstants.REMOVE_ENUM_VALUE_PREFIX).toString()));
         }
+
+        if (additionalProperties.containsKey(CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE)) {
```

Comment:
This is a confusing use of configuration. It's created in `Generate.java` as a CLI option, but then pulled from additionalProperties here. We should stick to one or the other (not both).

I think it makes sense to add it to `WorkflowSettings` because we're defining the workflow for geneation.

Let me know if you'd like help and I can contribute that directly to your branch.

Reply:
This has been changed to legacyDiscriminatorBehavior
I will update to the suggested integration